internal/github/github_test.go

```diff
@@ -1295,3 +1298,138 @@ func TestGetPRHeadSHA_Error(t *testing.T) {
 	_, err := ghClient.GetPRHeadSHA(ctx, "owner", "repo", 1)
 	require.Error(t, err)
 }
+
+func TestHandleResponseError(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	logger := testLogger
+	client := GHClient{logger: logger}
+
+	resetTime := time.Now().Add(time.Hour)
+
+	tests := []struct {
+		name           string
+		createResp     func() *github.Response
+		shouldBeError  bool
+		errorContains  []string
+		checkErrorType func(t *testing.T, err error)
```

Comment:
I'd suggest the following for these fields

- Replace them with one field
- Which is a pointer to an error
- If it's `nil`, that means no error is expected
- Otherwise, use `require.ErrorAs` on it (if you implement my suggestion from ☝️!)

Checking for types is a bit more robust than checking for strings - the types can be considered the contract, whereas if we change the strings because we like other ones more, all we'll have to do is update the tests to follow. That's one of the guidelines I use anyway when thinking about what to test. 😁